app/assets/javascripts/admin/addon/components/suspension-details.js

```diff
@@ -1,4 +1,44 @@
 import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import I18n from "I18n";
+import { observes } from "discourse-common/utils/decorators";
+
+const CUSTOM_REASON_KEY = "custom";
+
 export default Component.extend({
   tagName: "",
+  selectedReason: CUSTOM_REASON_KEY,
+  customReason: "",
+  reasonKeys: [
+    "not_listening_to_staff",
+    "consuming_staff_time",
+    "combatative",
+    "in_wrong_place",
+    "no_constructive_purpose",
+    CUSTOM_REASON_KEY,
+  ],
+
+  @discourseComputed("reasonKeys")
+  reasons(keys) {
+    return keys.map((key) => {
+      return { id: key, name: I18n.t(`admin.user.suspend_reasons.${key}`) };
+    });
+  },
+
+  @discourseComputed("selectedReason")
+  isCustomReason(selectedReason) {
+    return selectedReason === CUSTOM_REASON_KEY;
+  },
+
+  @observes("isCustomReason", "customReason", "selectedReason")
```

Comment:
We are trying to use less observers because they are hard to reason with and debug. In this case, you should be able to use `onChange` events from `combo-box` and `text-field` and only update the value when necessary.